File: 1544. Make The String Great.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

class Solution {
public:
    string makeGood(string s) {
        stack<char> st;

        for (int i = 0; i < s.length(); i++) {
            st.push(s[i]);
            char top = st.top(); st.pop();

            if (st.size() >= 1) {
                char secondTop = st.top();
                if (top >= 'A' && top <= 'Z') {
                    if (secondTop >= 'a' && secondTop <= 'z') {
                        if (top - 'A' == secondTop - 'a') st.pop();
                        else st.push(top);
                    } else {
                        st.push(top);
                    }
                } else {
                    if (secondTop >= 'A' && secondTop <= 'Z') {
                        if (top - 'a' == secondTop - 'A') st.pop();
                        else st.push(top);
                    } else {
                        st.push(top);
                    }
                }
            } else {
                st.push(top);
            }
        }

        string ans = "";
        while (!st.empty()) {
            ans += st.top();
            st.pop();
        }

        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 1544. Make The String Great.java
class Solution {
    public String makeGood(String s) {
        Stack<Character> stack = new Stack<Character>();
        for(int i = 0; i < s.length(); i++){
            stack.push(s.charAt(i));
            char top = stack.pop();
            
            if(stack.size() >= 1){
                char secondTop = stack.peek();
                if(top - 'A' >= 0 && top - 'A' <= 26){
                    if(secondTop - 'a' >= 0 && secondTop - 'a' <= 26){
                        if(top - 'A' == secondTop - 'a') stack.pop();
                        else stack.push(top);
                    }
                    else stack.push(top);
                }
                else{
                    if(secondTop - 'A' >= 0 && secondTop - 'A' <= 26){
                        if(top - 'a' == secondTop - 'A') stack.pop();
                        else stack.push(top);
                    }
                    else stack.push(top);
                }
            }
            else stack.push(top);
        }

        StringBuilder ans = new StringBuilder("");
        while(!stack.isEmpty()) ans.append(stack.pop());

        return ans.reverse().toString();
    }
}